addons/misra.py

```diff
@@ -2037,10 +2010,7 @@ def isRuleSuppressed(self, location, ruleNum):
         # Remove any prefix listed in command arguments from the filename.
         filename = None
         if location.file is not None:
-            if self.filePrefix is not None:
-                filename = remove_file_prefix(location.file, self.filePrefix)
-            else:
-                filename = location.file
+            filename = os.path.basename(location.file)
```

Comment:
This won't work if you want to use something like the below in the suppressions file.

src1/file.c
src2/file.c

That's why I did it by specifying a file prefix to remove.